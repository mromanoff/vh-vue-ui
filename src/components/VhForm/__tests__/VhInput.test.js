import { shallowMount } from '@vue/test-utils';
import VhInput from '../VhInput.vue';

describe('VhInput.vue', () => {
  let wrapper;

  it('renders default VhInput', () => {
    wrapper = shallowMount(VhInput, {
      propsData: {
        name: 'Test Name',
        label: 'Test Label',
      },
    });
    expect(wrapper.props().label).toBe('Test Label');
    expect(wrapper.props().labelHidden).toBe(false);
    expect(wrapper.props().name).toBe('Test Name');
    expect(wrapper.props().value).toBe('');
    expect(wrapper.props().disabled).toBe(false);
    expect(wrapper.props().state).toBe('pristine');
    expect(wrapper.props().message).toBe('');
    expect(wrapper.props().size).toBe('medium');
    expect(wrapper.props().theme).toBe('light');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders modified VhInput', () => {
    wrapper = shallowMount(VhInput, {
      propsData: {
        name: 'Test Name',
        label: 'Test Label',
        labelHidden: true,
        value: 'any',
        disabled: true,
        state: 'dirty',
        size: 'large',
        theme: 'dark',
        message: 'This is a test',
      },
    });
    expect(wrapper.props().label).toBe('Test Label');
    expect(wrapper.props().labelHidden).toBe(true);
    expect(wrapper.props().name).toBe('Test Name');
    expect(wrapper.props().value).toBe('any');
    expect(wrapper.props().disabled).toBe(true);
    expect(wrapper.props().state).toBe('dirty');
    expect(wrapper.props().message).toBe('This is a test');
    expect(wrapper.props().size).toBe('large');
    expect(wrapper.props().theme).toBe('dark');
    expect(wrapper).toMatchSnapshot();
  });
});
