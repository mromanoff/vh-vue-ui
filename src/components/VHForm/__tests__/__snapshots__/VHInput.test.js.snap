// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VHInput.vue renders default VHInput 1`] = `
<div class="Form-field Form-field--medium Form-field--light Form-field--pristine"><label for="Test Name" class="Form-label">
    Test Label
  </label> <input id="Test Name" class="Form-input">
  <p class="Form-message" style="display: none;">

  </p>
</div>
`;

exports[`VHInput.vue renders modified VHInput 1`] = `
<div class="Form-field Form-field--large Form-field--dark Form-field--dirty"><label for="Test Name" class="Form-label u-screenReaderOnly">
    Test Label
  </label> <input id="Test Name" disabled="disabled" class="Form-input Form-input--disabled">
  <p class="Form-message">
    This is a test
  </p>
</div>
`;
